In [3]:
num = 30

multiple assigments saved



In [8]:
x = y = z = 5 

In [9]:
a, b, c = 5, 10.1, "Great Lake"

re-assignment a variable

var1 = 25
var2 = "Arthur"
var2 = var1
var2

# Rules and Naming Conventions
Data Types
                    1. Numbers
                    2. Strings
                    3. Boolean
                    4. Lists
                    5. Tuples
                    6. Dictionaries 
                    7. Sets

There are four numeric Python data type:
a. int (signed integers)
b. float (floating point real values)
c. long (long integers, they can also be represented in octal and hexadecimal)
d. complex (complex numbers)



int	float	long	complex
10	54.4	012345L	2+5j
70	0.01	-987652L	1.18j
-150	1.626e16	10000000000000000	-10j
0x69	74.15	-4335685298777L	45j
-0x260	89.33	0348L	3e+26J

example of int

In [11]:
a = -5
type(a)

int

In [12]:
x = 1000
type(a)

int

example of floats

In [ ]:
a = 1.1
type(a)

In [13]:
x = -3.14
type(x)

float

In [14]:
y = 1.626e16
type(y)

float

example of long


In [15]:
x =999999999999999999999999999999999999999999999
type(x)

int

complex

In [16]:
x = 3+4j
type(x)

complex

In [17]:
y = 3e+4j
type(y)

complex

python string

In [18]:
string = "Introduction to Python"
string

'Introduction to Python'

The plus (+) sign is the string concatenation operator and the asterisk (*) is the repetition operator.

In [19]:
str1 = "Great"
str2 = "Learning"
str3 = str1+str2
str3 

'GreatLearning'

In [20]:
str3*3 

'GreatLearningGreatLearningGreatLearning'

You cannot use the minus(-) operator and the division(/) operator on strings.

Subsets of strings can be taken using the slice operator ([ ]). You can display a character from a string using its index in the string. Remember, indexing starts with 0.

In [21]:
string[5]

'd'

You can also display a burst of characters in a string using the slicing operator with indexes starting at 0 at the beginning of the string and working their way from -1 at the end.

In [22]:
string[13:22]

'to Python'

Strings are immutable. The following line of code will generate an error.

In [23]:
string[0] = 'P'

TypeError: 'str' object does not support item assignment

Python Boolean
In Python programming language, the boolean datatype is a primitive datatype having one of two values: True or False. It is often a result of some conditions.

In [24]:
var = True
type(var)

bool

In [25]:
print(10 > 9)

True


In [26]:
print(10 == 9)

False


In [27]:
print(10 < 9)

False


Type Conversion
Since Python is dynamically-typed, you may want to convert a value into another type. Python supports different type conversion functions like int(), float(), str() etc.

float()
It converts the passed value into a float.

In [28]:
float(11)

11.0

You can turn a boolean into an float.

In [29]:
float (True)

1.0

In [30]:
float(False)

0.0

You cannot turn a string into an float. It throws an error. However, if the string has only numbers, then you can.

In [31]:
float("abc")

ValueError: could not convert string to float: 'abc'

In [32]:
float("11")

11.0

In [33]:
float("2.24e-2")

0.0224

# You can also use ‘e’ to denote an exponential number
 # This number works even without the float() function

int 
Conversion from float to int will round-off the value (make it closer to zero)

In [34]:
int(11.7)

11

In [35]:
int(-11.7)

-11

You can also turn a boolean into an int.

In [36]:
int(True)

1

In [37]:
int(False)

0

You cannot turn a string into an int. It throws an error. However, if the string has only numbers, then you can.

In [38]:
int("GL")

ValueError: invalid literal for int() with base 10: 'GL'

In [39]:
int("11")

11

str() converts the value into a string.



In [40]:
str(2.24)

'2.24'

In [41]:
str(True)

'True'

In [42]:
str(False)

'False'

In [43]:
str(10)

'10'

Python print()
Syntax:
                     print(*objects, sep=' ', end='\n', file=sys.stdout, flush=False)
                print() Parameters:
                **objects** - object to be printed. * indicates that there may be more than one object
sep - objects are separated by sep. Default value: ' '
end - end is printed at last file - must be an object with write(string) method. If omitted it, sys.stdout will be used which prints objects on the screen. flush - If True, the stream is forcibly flushed. Default value: False




In [ ]:
print("Hello World")


In [44]:
x = 2
print("x =", x) 

x = 2


intilaizing a varibable here we pass 2 objects

print() with separator and end parameters
By default, python's print() function ends with a newline. This function comes with a parameter called 'end.' The default value of this parameter is '\n,' i.e., the new line character. You can end a print statement with any character or string using this parameter.
Sometimes you need to print one blank line in your Python program.

In [45]:
x = 2
a = 8
print("x =", x, end='\n\n\n')
print("a =", 8, end = 'j')

x = 2


a = 8j

In [46]:
x = 2
a = 8
print("x =", x, sep='$$')
print("a =", 8, sep = 'j')

x =$$2
a =j8


User Input
The Python input() function is used to read data from a standard input such as a keyboard and treats the received data as a string. When we enter a digit value, input() in python 3 stores it as a string value. So to convert string value to integer value, we will use the int() function just as shown below.

In [47]:
age = int(input("Please enter your age: "))
print(age)
type(age)

Please enter your age: 24
24


int

In [48]:
name = input("Please enter your name: ")
print(name)
type(name)

Please enter your name: mike
mike


str

When the input() function executes program flow will be stopped until the user has given input. The message displayed on the output screen to ask a user to enter an input value is optional i.e. the prompt, which will be printed on the screen is optional. Whatever you enter as input, the input function converts it into a string. if you enter an integer value still input() function converts it into a string. You need to explicitly convert it into an integer in your code using typecasting.

Arithmetic Operator
Arithmetic operators are used to perform mathematical operations like addition, subtraction, multiplication etc.

In [55]:
a = 12
b = 2
print('a + b =',a+b)
print('a * b =',a*b)
print('a % b =',a%b) 
print('a / b =',a/b) 
print('a // b =',a//b)
print('a ** b =',a**b)

a + b = 14
a * b = 24
a % b = 0
a / b = 6.0
a // b = 6
a ** b = 144


a = 12
b = 2
# '+' operator
print('a + b =',a+b)                  # Prints addition of a and b

# '-' operator
print('a - b =',a-b)                  # Prints subtraction of  a and b

# '*' operator
print('a * b =',a*b)                  # Prints multiplication of  a and b

# '%' operator
print('a % b =',a%b)                  # Prints remainder of division between a and b

# '/' operator
print('a / b =',a/b)                  # Prints division of  a and b

# '//' operator
print('a // b =',a//b)                # Prints the interger value of the division between a and b 

# '**' operator
print('a ** b =',a**b)                # Prints the result of a to the power b
a + b = 14
a - b = 10
a * b = 24
a % b = 0
a / b = 6.0
a // b = 6
a ** b = 144
Comparison Operator
Comparison Operator carries out the comparison between operands. They tell us whether an operand is greater than the other, lesser, equal, or a combination of those. It either returns True or False according to the condition.

a = 10
b = 20

# '>' operator
print('a > b  is',a>b)                 # Prints False since a is smaller than b

# '<' operator
print('a < b  is',a<b)                 # Prints True since a is smaller than b

# '==' operator
print('a == b  is',a==b)               # Prints False since a and b are not equal

# '!=' operator
print('a != b  is',a!=b)               # Prints True since a is not equal to b

# '>=' operator
print('a >= b  is',a>=b)               # Prints False since a is not smaller nor equal to b

# '<= y' operator
print('a <= b  is',a<=b)               # Prints True since a i

In [56]:
x = 5                                               # Assigns the value to the variable on the left
print("x = ", x)

# '+=' operator                 
x += 5 
print("Updated value of x after (x += 5) is", x)    # Adds the values on either side and assigns them to the expression on the left. x+=5 is the same as x=x+5

# '-=' operator
x -=2
print("Updated value of x after (x -=2) is", x)     # Subtracts 2 from the latest value of x and assigns it to x(which is 10)

# '*=' operator
x *= 2
print("Updated value of x after (x *= 2) is", x)    # Multiplies the current value of x with 2 and assigns the result to the same variable

# '/=' operator
x /= 4 
print('Updated value of x after (x /= 4) is', x)    # Divides the current value of x with 4 and assigns the result to the same variable

x =  5
Updated value of x after (x += 5) is 10
Updated value of x after (x -=2) is 8
Updated value of x after (x *= 2) is 16
Updated value of x after (x /= 4) is 4.0


i+=1 is the same as i=i+1, whereas i=+1 just means i=(+1)

Logical Operators
Logical operators are like conjunctions that you can use to combine more than one condition. We have three Python logical operator – and, or, and not that come under python operators.

'and' Operator
If the conditions on both the sides of the operator are true, then the expression as a whole is true.

In [57]:
x = 11
a = x>10 and x<20
print(a)

True


Here both the conditions are satisfied. Therefore we have True as the output.

In [58]:
x = 11
a = x>10 and x==12
print(a)

False


Here one of the condition is not satisfied. Therefore we have False as the output.

In [60]:
x = 11
a = x>10 or x==20
print(a)

True


Here one of the condition is satisfied. Therefore we have True as the output.

In [61]:
x = 11
a = x>12 or x==20
print(a)

False


None of the above condition is True. Therefore we have False as the output